### PR TITLE
GDB-6068 Implemented view for Smart Updates. 

### PR DESCRIPTION
**Pull request by sava-savov-ontotext**
The user will be able to see the list of already created templates, to edit and delete them, or create new template.

### CODE REVIEW

File: src/js/angular/sparql-template/controllers.js [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+
+    $scope.$on('$destroy', function (event) {
+        window.removeEventListener('beforeunload', showBeforeunloadMessage);
+        locationChangeListener();
+        $timeout.cancel(timer);
+    });
+
+    const defaultTabConfig = {
+        templateID: '',
+        query: 'PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n' +
```
Comment (desislava-hristova-ontotext) [inline src/js/angular/sparql-template/controllers.js:138]: The default query should be something more generic like:
`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX ex: <http://example.com#>
DELETE {
    ?subject ex:myPredicate ?oldValue .
} INSERT {
    ?subject ex:myPredicate ?newValue .
} WHERE {
    ?id rdf:type ex:MyType .
    ?subject ex:isRelatedTo ?id .
}`

File: src/js/lib/angularjs/1.3.8/angular.js [hunk 2]
```diff
@@ -5678,7 +5678,7 @@ function $CacheFactoryProvider() {
  * Adding via the `script` tag:
  *
  * ```html
- *   <script type="text/ng-template" id="templateId.html">
```
Comment (desislava-hristova-ontotext) [inline src/js/lib/angularjs/1.3.8/angular.js:5681]: You should not modify the angular.js file.

File: src/js/angular/sparql-template/controllers.js [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+    .controller('SparqlTemplateCreateCtrl', SparqlTemplateCreateCtrl);
+
+SparqlTemplatesCtrl.$inject = ['$scope', '$repositories', 'SparqlTemplatesRestService', 'toastr', 'ModalService'];
+
+function SparqlTemplatesCtrl($scope, $repositories, SparqlTemplatesRestService, toastr, ModalService) {
+
+    $scope.getSparqlTemplates = function () {
+        // Only do this if there is an active repo that isn't an Ontop repo.
+        // Ontop repos doesn't support update operations.
+        if ($repositories.getActiveRepository() && !$repositories.isActiveRepoOntopType()) {
```
Comment (desislava-hristova-ontotext) [inline src/js/angular/sparql-template/controllers.js:27]: What about FedX repos? I guess it should not be supported also.

**Review by desislava-hristova-ontotext — CHANGES_REQUESTED** [review]

**Comment by sava-savov-ontotext** [comment]
Jenkins retry build

**Review by desislava-hristova-ontotext — APPROVED** [review]

**Pull request merged**